### DISCUSSION TITLE
[Feature Request]: Change LARS behaviour to reset document context on New Chat

### DISCUSSION THREAD

**Discussion by ShaswatPanda** [abgulati/LARS#13] [Ideas]
Currently, LARS has a default behavior of making ALL the documents you ever uploaded persist across all conversations. (Which would be helpful in some use cases) However, this also happens even if you want answers from only a specific document.

Wanted to know if it would be possible to reset the vectorDB every time we start a new chat as a default behavior? Or maybe each conversation should have its own vectorDB inside its history? That way conversations would stay specific and isolated, just like how it is in most applications like ChatGPT or Gemini.

**Comment by abgulati** [comment]
Hi @ShaswatPanda 

Being a RAG application, LARS is built to reference it's vector database when responsding to queries. In ChatGPT or Gemini, uploaded docuemts are taken into context but not embedded into a vector DB, thus making their approach fundamentally different from RAG.

That said, LARS does allow you to disable RAG, or even reset the vector db via the UI. Resetting does not delete the vector db on disk, and you're free to manually manage the version of the vector db used via the config.json file. This is more flexibility than a typical RAG application, but has been built to address requirements such as these.

Since a unified vector database is maintained, as is required for a accurate semantic serach that covers searches for the best data in a knowledge store when responding to queries, restricting the search to specific documents in a single vectorDB is not possible at this time and the aforementioned approaches must be taken to accomodate special requirements. 

In the future, I will likely explore vectorDB alternatives to ChromaDB and if such funtionality is available, it will be introduced in LARS in due course. Being an open-source project, the community is welcome to add contributions for desired features too.

Keep an eye on the project repo and development roadmap for updates!
